### PR TITLE
Vastly simplify taking final greenery action, making it philares-compatible

### PR DESCRIPTION
**Pull request by kberg**
Instead of relying on knowing that Philares is involved and shunting it into the existing code, this makes the assumption that if a player has a `waitingFor` they just aren't done. It also abandons precomputing whether a player may place greeneries during the phase start in favor of determining whether the player can place a greenery when it's that player's turn.

It's kind of unfortunate that this doesn't just use the existing code about active players and transitions, making it understandably difficult to parse and understand. I hope this code is easier to understand, made even easier by comments. Add more comments as need be. I will probably add more.

### CODE REVIEW

**Comment by kberg** [comment]
I don't want to merge this just yet -- do some more play testing first. But see if you can spot anything awful.

File: src/Game.ts [hunk 1]
```diff
@@ -1051,51 +1052,40 @@ export class Game implements ISerializable<SerializedGame> {
...
-    const players: Player[] = [];
-
-    this.players.forEach((player) => {
+  // Well, this isn't just "go to the final greenery placement". It finds the next player
+  // who might be able to place a final greenery.
+  // Rename to takeNextFinalGreeneryAction?
+  public gotoFinalGreeneryPlacement(): void {
+    // this.getPlayers returns in turn order -- a necessary rule for final greenery placement.
+    const players = this.getPlayers();
+    for (let idx = 0; idx < players.length; idx++) {
```
Comment (kberg) [inline src/Game.ts:1074]: An old-fashioned for loop! How antiquated. But sometimes lambdas are harder to read. This is just such a case.

Comment (bafolts) [reply src/Game.ts:1075]: If you don't need the `idx` you can write this as `for (const player of players)`.

File: src/Game.ts [hunk 2]
```diff
@@ -1051,51 +1052,41 @@ export class Game implements ISerializable<SerializedGame> {
...
+      const player = players[idx];
+      if (this.donePlayers.has(player.id)) {
+        continue;
+      }
       if (this.canPlaceGreenery(player)) {
-        players.push(player);
+        this.startFinalGreeneryPlacement(player);
+        return;
+      } else if (player.getWaitingFor() !== undefined) {
```
Comment (bafolts) [inline src/Game.ts:1083]: This seems different from what we had before. Is this part of the change that now if any player who is not done has a waiting for we wait for that action?

File: src/Game.ts (hunk 1 above)
Comment (kberg) [reply src/Game.ts]: Yes, I keep forgetting about that expression form. Thanks!

File: src/Game.ts (hunk 2 above)
Comment (kberg) [reply src/Game.ts:1081]: Correct. If you are waiting for the next possible player to do something, then they're not actually done for the game, yet.

**Review by bafolts — APPROVED** [review]

**Pull request merged**